File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people);
        ArrayList<Integer> list = new ArrayList<>();
        for (int i : people) {
            list.add(i);
        }

        while (!list.isEmpty()) {
            int lastIdx = list.size() - 1;
            if (lastIdx == 0) {
                return answer+1;
            } else {
                if (limit - list.get(lastIdx) >= list.get(0)) {
                    list.remove(0);
                    list.remove(lastIdx-1);
                }
                else {
                    list.remove(lastIdx);
                }
                
            }
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    var answer = 0;
    arr = people.sort((a,b)=> a-b)
    let left = 0
    let right = arr.length-1
    
    while(left <= right) {
        if(limit >= arr[left]+arr[right]) {
            left++
        }
        right--
        
        answer++
    }
    return answer;
}